src/components/ProductCarousel/ProductCarousel.tsx

```diff
@@ -0,0 +1,96 @@
+import React, {useCallback, useRef, useState} from 'react';
+import {Dimensions, View, ViewStyle} from 'react-native';
+import {GestureHandlerRootView} from 'react-native-gesture-handler';
+import {SharedValue} from 'react-native-reanimated';
+import Carousel, {ICarouselInstance} from 'react-native-reanimated-carousel';
+
+import {Button, Dots} from './components';
+
+import {styles} from './ProductCarousel.styles';
+
+type CarouselComponents = {
+  Dots: typeof Dots;
+  Button: typeof Button;
+};
+
+type ProductCarouselProps = {
+  children: JSX.Element[];
+  slideStyle?: ViewStyle;
+  containerStyle?: ViewStyle;
+  height?: number;
+  width?: number;
+  pagination?: boolean;
+  withControls?: boolean;
+};
+
+export const ProductCarousel: React.FC<ProductCarouselProps> &
+  CarouselComponents = ({
+  slideStyle,
+  children,
+  height,
+  width,
+  pagination = true,
+  containerStyle,
+  withControls = true,
+}) => {
+  console.log(children);
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const screenWidth = Dimensions.get('window').width;
```

Comment:
you can store it in constants, it will be not recalculated on the window size change anyway